lib/src/source/git.dart

```diff
@@ -253,9 +302,22 @@ class GitSource extends CachedSource {
   /// by [id] on the effective ref of [id].
   ///
   /// This assumes that the canonical clone already exists.
-  Future<String> _getRev(PackageId id) {
-    return git.run(["rev-list", "--max-count=1", _getEffectiveRef(id)],
-        workingDir: _repoCachePath(id)).then((result) => result.first);
+  Future<String> _getRev(PackageId id) async {
```

Comment:
If you're changing `_getRev` rather than `_getEffectiveRef` (which makes sense), you'll need to update `downloadToSystemCache` to use the former so that it behaves properly with `v`-versions.